### ISSUE TITLE
extremely slow to read sign fields and certificate from a simple one page PDF

### ISSUE THREAD

**Issue by insinfo** [syncfusion/flutter-widgets#1952]
### Bug description

extremely slow to read sign fields and certificate from a simple one page PDF
2 minutes and 23 seconds to read 

### Steps to reproduce

```dart
import 'dart:io';
import 'package:syncfusion_flutter_pdf/pdf.dart';
import 'package:syncfusion_flutter_pdf/src/pdf/implementation/general/pdf_collection.dart';

void main(List<String> args) async {
  final inputBytes =
      await File('acordo_extrajudicial_assinado.pdf').readAsBytes();
      
  print('${DateTime.now()} main inputBytes');
  //Loads PDF file.
  final document = PdfDocument(inputBytes: inputBytes);

  print('${DateTime.now()} main Loads PDF file');

  final signFields = PdfObjectCollectionHelper.getHelper(document.form.fields)
      .list
      .where((field) => field is PdfSignatureField)
      .toList();

  print(
      '${DateTime.now()} main Loads signFields ${signFields.length}');

  if (signFields.isNotEmpty) {
    for (final signField in signFields) {
      final signItem = signField as PdfSignatureField;

      print(
          '${DateTime.now()} main signItem ${signItem.signature}');
      print(
          '${DateTime.now()} has signature and certificate: ${signItem.signature?.certificate}');
    }
  } else {
    print('PDF without digital signature');
  }
}

```

[acordo_extrajudicial_assinado.pdf](https://github.com/user-attachments/files/15979983/acordo_extrajudicial_assinado.pdf)

### log
```
2024-06-25 19:20:18.296638 main inputBytes
2024-06-25 19:20:18.483179 main Loads PDF file
2024-06-25 19:20:18.561842 main Loads signFields 1
2024-06-25 19:20:18.561842 main signItem Instance of 'PdfSignature'
2024-06-25 19:22:42.351617 has signature and certificate: null
```

</details>

### On which target platforms have you observed this bug?

Windows

### Flutter Doctor output

Dart SDK version: 3.2.1 (stable) (Wed Nov 22 08:59:13 2023 +0000) on "windows_x64"

**Comment by jeyalakshmit** [comment]
Hi @insinfo ,

We were able to reproduce the reported issue with the provided details on our end, and currently, we are analyzing it. We will provide further details on June 28, 2024.

Regards,
Jeyalakshmi T

**Comment by insinfo** [comment]
I tested it with another file, but it doesn't display the certificate information, the certificate is null

```dart

void main(List<String> args) async {
  final inputBytes =
      await File('Arquivo teste de assinatura digital - Assinado.pdf')
          .readAsBytes();

  print('${DateTime.now()}  inputBytes');
  //Loads PDF file.
  final document = PdfDocument(inputBytes: inputBytes);

  print('${DateTime.now()} loads PDF file');

  final signFields = PdfObjectCollectionHelper.getHelper(document.form.fields)
      .list
      .where((field) => field is PdfSignatureField)
      .map((e) => e as PdfSignatureField)
      .toList();

  print('${DateTime.now()} Loads signFields ${signFields.length}');

  if (signFields.isNotEmpty) {
    for (final signField in signFields) {
      final signItem = signField;
      // Check if field is signed.
      print('${DateTime.now()}  has signature ${signItem.isSigned}');
      print('${DateTime.now()}  signItem ${signItem.signature}');
      print('${DateTime.now()}  signedName ${signItem.signature?.signedName}');
      print(
          '${DateTime.now()}  contactInfo ${signItem.signature?.contactInfo}');
      print('${DateTime.now()}  signedDate ${signItem.signature?.signedDate}');
      print(
          '${DateTime.now()}  certificate: ${signItem.signature?.certificate}');
    }
  } else {
    print('PDF without digital signature');
  }

// Dispose the document.
  document.dispose();
}

```

![image](https://github.com/syncfusion/flutter-widgets/assets/12227024/9c47d5ad-0267-42c6-93d2-2f1ef973665c)
[Arquivo teste de assinatura digital - Assinado.pdf](https://github.com/user-attachments/files/15991396/Arquivo.teste.de.assinatura.digital.-.Assinado.pdf)

**Comment by AnanthaGokulaRaman** [comment]
Hi @insinfo,

|     |     |
| --- | --- |
| extremely slow to read sign fields and certificate from a simple one page PDF  <br>2 minutes and 23 seconds to read | We have confirmed the issue ”**Performance issue occurs when reading the signature field from a specific PDF document**” as a defect in our product and fix will be included in our upcoming weekly release, which will be available on July 9, 2024.<br><br>Use the below feedback link to track the status of the reported bug.<br><br>[Performance issue occurs when reading the signature field from a specific PDF document in Flutter \| Feedback Portal (syncfusion.com)](https://www.syncfusion.com/feedback/59097/performance-issue-occurs-when-reading-the-signature-field-from-a-specific-pdf)<br><br>Disclaimer: “Inclusion of this solution in the weekly release may change due to other factors including but not limited to QA checks and works reprioritization.” |
| &nbsp;tested it with another files, but it doesn't display the certificate information, the certificate is null | Currently we do not have support to read the certificate data from a loaded signature. We do not have any immediate plans to implement this feature and we will implement this support in any of our upcoming releases.<br><br>Please use the below feedback link to track the status of the feature.<br><br>[Support to validate the digital signatures in an existing PDF document in Flutter \| Feedback Portal (syncfusion.com)](https://www.syncfusion.com/feedback/25198/support-to-validate-the-digital-signatures-in-an-existing-pdf-document) |

Regards,
Anantha Gokula Raman J

**Comment by jeyalakshmit** [comment]
Hi @insinfo ,

The fix for the reported issue "Performance issue occurs when reading the signature field from a specific PDF document" has been included in our weekly release v26.1.41.

Kindly use the following link to refer to the latest release version,

[syncfusion_flutter_pdf package - All Versions (pub.dev)](https://pub.dev/packages/syncfusion_flutter_pdf/versions)

Root cause: To read the signature data, we decode the hex string using our parser. In this case, decoding a large hex string caused a performance issue.
